controllers/applicationvolumereplication_controller.go

```diff
@@ -231,14 +219,155 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 	r.Log.Info(fmt.Sprintf("AVR (%+v)", avr))
 
 	logger.Info(fmt.Sprintf("Processed %d/%d subscriptions. AVR Status Decisions %d. Requeue? %v",
-		len(a.instance.Status.Decisions), len(subscriptionList.Items), len(avr.Status.Decisions), requeue))
+		len(a.instance.Status.Decisions), len(subList.Items), len(avr.Status.Decisions), requeue))
 
 	done := !requeue
 	r.Callback(a.instance.Name, done)
 
 	return ctrl.Result{Requeue: requeue}, nil
 }
 
+func (r *ApplicationVolumeReplicationReconciler) getSubscriptionList(ctx context.Context,
+	avr *rmn.ApplicationVolumeReplication) (*subv1.SubscriptionList, error) {
+	subscriptionList := &subv1.SubscriptionList{}
+	listOptions := []client.ListOption{
+		client.InNamespace(avr.Namespace),
+		client.MatchingLabels(avr.Spec.SubscriptionSelector.MatchLabels),
+	}
+
+	err := r.Client.List(ctx, subscriptionList, listOptions...)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			r.Log.Error(err, "failed to find subscription list", "namespace", avr.Namespace)
+
+			return nil, errorswrapper.Wrap(err, "No subscriptions found")
+		}
+
+		return nil, errorswrapper.Wrap(err, "failed to list subscriptions")
+	}
+
+	err = r.ensureValidSubscriptions(avr, subscriptionList)
+	if err != nil {
+		r.Log.Error(err, "unable to ensure subscriptions have the same placement rule", "namespace", avr.Namespace)
+
+		return nil, err
+	}
+
+	return subscriptionList, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) ensureValidSubscriptions(
+	avr *rmn.ApplicationVolumeReplication, subscriptions *subv1.SubscriptionList) error {
+	// Make sure that all subscriptions in subscription list have same placementRule
+	if err := r.ensureSubsHaveSamePlacementRule(avr, subscriptions); err != nil {
+		return err
+	}
+
+	// Make sure clusters referenced by the placementRule are in the ClusterPeers for this AVR.
+	return r.ensurePlRuleClustersInClusterPeers(avr)
+}
+
+func (r *ApplicationVolumeReplicationReconciler) ensureSubsHaveSamePlacementRule(
+	avr *rmn.ApplicationVolumeReplication, subscriptions *subv1.SubscriptionList) error {
+	if len(subscriptions.Items) == 0 {
+		return fmt.Errorf("empty subscriptionList for avr %s", avr.Name)
+	}
+
+	for _, sub := range subscriptions.Items {
+		if sub.Spec.Placement != nil && sub.Spec.Placement.PlacementRef != nil {
```

Comment:
Should fail for nil Placements as well, currently nil placements are ignored

Reply:
done